src/random.cr

```diff
@@ -277,6 +290,115 @@ module Random
     end
   end
 
+  # Fills a given slice with random bytes.
+  #
+  # ```
+  # slice = Bytes.new(4) # => [0, 0, 0, 0]
+  # Random.new.random_bytes(slice)
+  # slice # => [217, 118, 38, 196]
+  # ```
+  def random_bytes(buf : Bytes) : Nil
```

Comment:
The one method I added is this one: a generic `random_bytes(buf)` that depends on `next_u` size, to consume as few numbers as possibles from the PRNG.

I'm not very confident in the filling remaining bytes part (when `buf.size` isn't a multiple of `sizeof(next_u)`), and would like some review.

Reply:
:-( I see: method should be called differently. `random_bytes_native`?

Reply:
This code is in the base `Random` class. PRNGs are free to override `random_bytes`, the only bad thing is that they'd have to reimplement the buffer filling logic.

Reply:
@ysbaddaden The logic seems good to me :-)

Reply:
@asterite, but were you reviewing the old code that these comments are on, or the new code?

Reply:
@oprypin I only see one code, the same I see when I check out the code:

```crystal
  def random_bytes(buf : Bytes) : Nil
    ptr = buf.to_unsafe
    finish = buf.to_unsafe + buf.size

    while ptr < finish
      random = next_u
      rand_ptr = pointerof(random).as(UInt8*)
      if IO::ByteFormat::SystemEndian != IO::ByteFormat::LittleEndian
        rand_ptr.to_slice(sizeof(typeof(next_u))).reverse!
      end
      rand_ptr.copy_to(ptr, {finish - ptr, sizeof(typeof(next_u))}.min)
      ptr += sizeof(typeof(next_u))
    end
  end
```